crates/cranelift/src/stack_switching/instructions.rs

```diff
@@ -0,0 +1,1910 @@
+use cranelift_codegen::ir::BlockArg;
+use itertools::{Either, Itertools};
+
+use cranelift_codegen::ir::condcodes::*;
+use cranelift_codegen::ir::types::*;
+use cranelift_codegen::ir::{self, MemFlags};
+use cranelift_codegen::ir::{Block, BlockCall, InstBuilder, JumpTableData};
+use cranelift_frontend::FunctionBuilder;
+use wasmtime_environ::{PtrSize, TagIndex, TypeIndex, WasmResult, WasmValType};
+
+// TODO(frank-emrich) This is the size for x64 Linux. Once we support different
+// platforms for stack switching, must select appropriate value for target.
+pub const CONTROL_CONTEXT_SIZE: usize = 24;
```

Comment:
Can we put this into a helper function something like this:

```rust
fn control_context_size(triple: &target_lexicon::Triple) -> WasmResult<usize> {
    match (triple.architecture, triple.operating_system) {
        (target_lexicon::Architecture::X86_64, target_lexicon::OperatingSystem::Linux) => Ok(24),
        _ => Err(wasm_unsupported!("stack switching is not supported on {triple}")),
    }
}
```

This will future proof the code and give us a nice place to slot in updates when we expand platform support.